src/utils/TokenUtils.ts

```diff
@@ -0,0 +1,281 @@
+    if (token.adminKey) {
+      transaction.setAdminKey(getPrivateKey(token.adminKey));
+    }
+    if (token.kycKey) {
+      transaction.setKycKey(getPrivateKey(token.kycKey));
+    }
+    if (token.freezeKey) {
+      transaction.setFreezeKey(getPrivateKey(token.freezeKey));
+    }
+    if (token.pauseKey) {
+      transaction.setPauseKey(getPrivateKey(token.pauseKey));
+    }
+    if (token.wipeKey) {
+      transaction.setWipeKey(getPrivateKey(token.wipeKey));
+    }
+    if (token.feeScheduleKey) {
+      transaction.setFeeScheduleKey(getPrivateKey(token.feeScheduleKey));
+    }
+  }
+
+  /**
+   * Sets the optional properties of the token create transaction.
+   * @param transaction The transaction to set the properties on.
+   * @param token The properties of the token to create.
+   */
+  private static setOptionalProperties(transaction: TokenCreateTransaction, token: ITokenProps): void {
+    if (token.freezeDefault !== undefined) {
+      transaction.setFreezeDefault(token.freezeDefault);
+    }
+    if (token.autoRenewAccountId) {
+      transaction.setAutoRenewAccountId(token.autoRenewAccountId);
+    }
+    if (token.expirationTime) {
+      transaction.setExpirationTime(new Date(token.expirationTime));
+    }
+    if (token.autoRenewPeriod >= 2_592_000 && token.autoRenewPeriod <= 8_000_000) {
+      transaction.setAutoRenewPeriod(token.autoRenewPeriod);
+    }
+    if (token.tokenMemo) {
+      transaction.setTokenMemo(token.tokenMemo);
+    }
+    if (token.customFees) {
+      // TODO: Test this
+      transaction.setCustomFees(token.customFees.map(CustomFee._fromProtobuf));
+    }
+  }
+
+  private static validateTokenProperties(token: ITokenProps) {
+    this.assertTruthy(token.tokenName, 'Token name is required');
+    this.assertTruthy(token.tokenSymbol, 'Token symbol is required');
+    this.assertTruthy(token.tokenType, 'Token type is required');
+    this.assertTruthy(token.supplyType, 'Supply type is required');
+    // If the token type is NON_FUNGIBLE_UNIQUE,
+    // the initial supply must be 0 and decimals must be undefined
+    if (token.tokenType === TokenType.NonFungibleUnique.toString()) {
+      this.assertFalsy(token.initialSupply, 'Initial supply must be 0 or undefined for non-fungible tokens');
+      this.assertFalsy(token.decimals, 'Decimals must be 0 or undefined for non-fungible tokens');
+    } else {
+      this.assertTruthy(token.initialSupply, 'Initial supply is required for fungible tokens');
+      this.assertTruthy(token.decimals, 'Decimals is required for fungible tokens');
+    }
+    // If the token supply type is FINITE, the max supply must be provided
+    if (token.supplyType === TokenSupplyType.Finite.toString()) {
+      this.assertTruthy(token.maxSupply, 'Max supply is required for finite supply tokens');
+    } else {
+      this.assertFalsy(token.maxSupply, `Max supply must be undefined for infinite supply tokens, was ${token.maxSupply}`);
```

Comment:
I think if we have INFINITE supply, we should assume that the user would also need a supply key to mint new tokens. Maybe check for that as well?

Reply:
The operator key is used as supply key by default, if one is not provided by the user. 

We still give them freedom to decide if they want to use a different supply key, but if they don't provide one, the field is not required and it will auto-populate it with the operator key.